pkg/interface/src/components/ErrorBoundary.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { Component } from 'react';
+import { RouteComponentProps, withRouter } from 'react-router-dom';
+
+
+class ErrorBoundary extends Component<
+  RouteComponentProps,
+  { error?: Error }
+> {
+  constructor(props) {
+    super(props);
+    this.state = { error: undefined };
+    const { history } = this.props;
+    history.listen((location, action) => {
+      if (this.state.error) {
```

Comment:
How does this handle intra-app navigations? i.e. if the app crashes, I might click back which might keep me in the app that crashed?

Reply:
If you 1) visit a working page, 2) visit a broken page and see the error and 3) click on the sidebar to a new page then 4) click the back button, you'll see the same error, yes. But it will have been recomputed because you navigated to a new page, the router calculated where it was supposed to be, and ran into an error *again*. 
The error boundary exists to *keep* the app from crashing — just to catch faulty components. If something outside the boundary crashes then tough luck — WSOD 4 U.